src/pc/PlatformDeviceControl.c

```diff
@@ -564,6 +580,46 @@ int usbPlatformConnect(const char *devPathRead, const char *devPathWrite, void *
 #endif  /*USE_USB_VSC*/
 }
 
+int tcpipPlatformConnect(const char *devPathRead, const char *devPathWrite, void **fd)
+{
+#if defined(USE_TCP_IP)
+    sockfd = socket( AF_INET, SOCK_STREAM, 0 );
+    if( sockfd < 0 )
+    {
+        perror("socket");
+        return -1;
+    }
+
+    struct sockaddr_in serv_addr;
+    memset( &serv_addr, 0, sizeof( serv_addr ) );
+
+    int len = strlen( devPathWrite );
+    char devPathWriteBuff[len];
+    strcpy( devPathWriteBuff, devPathWrite );
+
+    char* serv_ip = strtok( devPathWriteBuff, ":" );
+    char* serv_port = strtok( NULL, ":" );
+
+    serv_addr.sin_family = AF_INET;
+    serv_addr.sin_port = htons( atoi( serv_port ) );
+
+    if( inet_pton( AF_INET, serv_ip, &serv_addr.sin_addr ) <= 0 )
+    {
+        perror("inet_pton");
+        return -1;
+    }
+
+    if( connect( sockfd, ( struct sockaddr * ) &serv_addr, sizeof( serv_addr ) ) < 0 )
+    {
+        perror("connect");
+        return -1;
+    }
+
```

Comment:
Must the socket be closed before returning in these cases?

Reply:
yes, I think so. will fix this